### PR TITLE
[11.0][IMP] base_tier_validation

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
This PR has the following objectives:
* Remove from base module the python expression fields to define the domain
* Add python expression to define domain and reviewers in a separate module (base_tier_validation_formula)
* Add in the systray a menu that shows the pending reviews the user has at the moment
* Add migration script to ensure a correct versions migration

cc ~ @etobella @jbeficent @eficent

### CODE REVIEW

**Comment by AdriaGForgeFlow** [comment]
Missing tests for base_tier_validation_formula and systray icon auto update on tier review changes

File: base_tier_validation/migrations/11.0.1.2.0/post-migrate.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from openupgradelib.openupgrade import migrate
+
+
+@migrate()
+def migrate(env, version):
+    module_ids = env['ir.module.module'].search([
+        ('name', '=', 'base_tier_validation_formula'),
+        ('state', '=', 'uninstalled')
+    ])
+    module_ids.sudo().button_install()
```
Comment (MiquelRForgeFlow) [inline base_tier_validation/migrations/11.0.1.2.0/post-migrate.py]: if module_ids:

File: base_tier_validation/static/description/index.html [hunk 2]
```diff
@@ -3,13 +3,13 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-<meta name="generator" content="Docutils 0.14: http://docutils.sourceforge.net/" />
+<meta name="generator" content="Docutils 0.12: http://docutils.sourceforge.net/" />
```
Comment (MiquelRForgeFlow) [inline base_tier_validation/static/description/index.html:6]: why?

File: base_tier_validation/static/src/xml/systray.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            <a class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Reviews" href="#">
+                <i class="fa fa-pencil-square-o"/> <span class="o_notification_counter badge"/>
+            </a>
+            <ul class="o_mail_navbar_dropdown dropdown-menu" role="menu">
+                <li class="o_mail_navbar_dropdown_channels"/>
+            </ul>
+        </li>
+    </t>
+
+</templates>
```
Comment (MiquelRForgeFlow) [inline base_tier_validation/static/src/xml/systray.xml]: end with blank line

File: base_tier_validation/views/assets_backend.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+<odoo>
+    <template id="assets_backend" name="mail assets"
+              inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <script type="text/javascript" src="/base_tier_validation/static/src/js/systray.js"/>
+
+            <link rel="stylesheet" href="/base_tier_validation/static/src/less/systray.less" type="text/less"/>
+        </xpath>
+    </template>
+</odoo>
```
Comment (MiquelRForgeFlow) [inline base_tier_validation/views/assets_backend.xml]: end with blank line

File: base_tier_validation/__manifest__.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Base Tier Validation",
     "summary": "Implement a validation process based on tiers.",
-    "version": "11.0.1.1.0",
+    "version": "11.0.1.2.0",
```
Comment (MiquelRForgeFlow) [inline base_tier_validation/__manifest__.py]: 11.0.2.0.0 instead, remember that you are changing dependencies and are adding migration scripts

**Comment by etobella** [comment]
I think that a dependency to bus is required

**Comment by etobella** [comment]
Great job! I think it is a great improvement! :smile:

**Review by JordiBForgeFlow — APPROVED** [review]
@ageficent can you squash the commits? This is ready to merge!

**Review by etobella — APPROVED** [review]

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
not ready, we found an issue, hang on a second! :D

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

File: base_tier_validation/models/tier_review.py [hunk 6]
```diff
@@ -38,6 +39,7 @@ class TierReview(models.Model):
     requested_by = fields.Many2one(
         comodel_name="res.users",
     )
+    reviewed_date = fields.Datetime(string='Validation Date')
 
     @api.multi
     @api.depends('reviewer_id', 'reviewer_group_id', 'reviewer_group_id.users')
```
Comment (etobella) [inline base_tier_validation/models/tier_review.py]: Can you change this depends for a function? Somethink like `lambda self: self.reviewer_depend_fields()`

This way, it is more secure to add more reviewer options.

**Review by etobella — COMMENTED** [review]
I will make a PR to your branch in order to review this change.

**Comment by rousseldenis** [comment]
@ageficent With this change being merged, you introduced errors in main branches in other repos (as some other modules depend on this one).

IMHO, such changes couldn't happen in already merged modules but should be done in a next major version. Who will take care of fixing all dependent modules ?

Example : https://github.com/OCA/stock-logistics-warehouse/tree/11.0/stock_request_tier_validation @lreficent @hveficent 

I remind you the OCA stable policy (yes, there is one :smile:  ) https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/oca_module_lifecycle_development_status.rst#stable-modules

**Comment by HviorForgeFlow** [comment]
@rousseldenis I do, please review :)
https://github.com/OCA/purchase-workflow/pull/654
https://github.com/OCA/stock-logistics-warehouse/pull/584

**Comment by rousseldenis** [comment]
> @rousseldenis I do, please review :)
> [OCA/purchase-workflow#654](https://github.com/OCA/purchase-workflow/pull/654)
> [OCA/stock-logistics-warehouse#584](https://github.com/OCA/stock-logistics-warehouse/pull/584)

I will do.

That does not affect what I said as people may have also inherited somewhere else. Please keep that in mind to avoid future troubles. Thanks

**Comment by AdriaGForgeFlow** [comment]
> That does not affect what I said as people may have also inherited somewhere else. Please keep that in mind to avoid future troubles. Thanks

I will have this in mind to avoid future troubles. Thanks for the insights :+1:

**Pull request merged**